bft/src/main.rs

```diff
@@ -1,14 +1,7 @@
 mod cli;
 
 fn main() {
-    match cli::run_bft() {
-        Err(e) => {
-            println!(
-                "{}: Error: {}",
-                std::env::args().nth(0).unwrap(),
-                e.to_string()
-            )
-        }
-        Ok(_) => (),
+    if let Err(e) = cli::run_bft() {
+        println!("{}: Error: {}", std::env::args().next().unwrap(), e,)
```

Comment:
If you weren't missing the non-zero exit code I mentioned; this whole commit would have been unnecessary